frontend/ui/screensaver.lua

```diff
@@ -148,15 +148,27 @@ function Screensaver:expandSpecial(message, fallback)
             time_left_chapter = self:_calcAverageTimeForPages(ui.toc:getChapterPagesLeft(currentpage) or doc:getTotalPagesLeft(currentpage))
             time_left_document = self:_calcAverageTimeForPages(doc:getTotalPagesLeft(currentpage))
         end
-        percent = Math.round((currentpage * 100) / totalpages)
+        if currentpage == 1 then
+            percent = 0
+        elseif currentpage == totalpages then
+            percent = 100
+        else
+            percent = Math.round(Math.clamp(((currentpage * 100) / totalpages), 1, 99))
+        end
         props = ui.doc_props
     elseif DocSettings:hasSidecarFile(lastfile) then
         -- If there's no ReaderUI instance, but the file has sidecar data, use that
         local doc_settings = DocSettings:open(lastfile)
         totalpages = doc_settings:readSetting("doc_pages") or totalpages
         percent = doc_settings:readSetting("percent_finished") or percent
         currentpage = Math.round(percent * totalpages)
-        percent = Math.round(percent * 100)
+        if currentpage == 1 then
+            percent = 0
+        elseif currentpage == totalpages then
+            percent = 100
+        else
+            percent = Math.round(Math.clamp(((currentpage * 100) / totalpages), 1, 99))
```

Comment:
You've got original `percent` in line 163, no need to recalculate it.

Reply:
Since I was puzzled for a second, that means:
```suggestion
            percent = Math.round(Math.clamp(percent, 1, 99))
```